src/Select/Select.component.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import Select from 'react-select';
+import styled from 'styled-components';
+
+type PropsThemeOnly = {
+  /**
+   * From theme provider
+   *
+   * @default defaultTheme
+   **/
+  theme?: any;
+};
+
+type PropsThemeAndInputSize = PropsThemeOnly & {
+  /**
+   * Select Input Size
+   *
+   * @default 'md'
+   **/
+  inputSize: string;
+};
+
+type Props = PropsThemeAndInputSize & {
+  /**
+   * The ID of the control
+   *
+   * @default null
+   **/
+  id: string;
+  /**
+   * Specify if the tab is disabled
+   *
+   * @default false
+   **/
+  disabled?: boolean;
+  /**
+   * Specify whether the control is currently invalid
+   *
+   * @default false
+   **/
+  invalid?: boolean;
+  /**
+   * Provide the text that is displayed when the control is in an invalid state
+   */
+  invalidText?: string;
+  /**
+   * The list of options available
+   *
+   * @default null
+   **/
+  options?: any;
+  /**
+   * Specify the control's selected option
+   *
+   * @default null
+   **/
+  selectedOption?: string;
+  /**
+   * Any props that should be passed directly to the third-
+   * party react-select control.
+   *
+   * @default null
+   **/
+  controlSpecificProps?: any;
+};
+
+const SDiv = styled.div`
+  background: ${(props: PropsThemeAndInputSize) => props.theme.input.background}
+  border: 1px solid ${(props: PropsThemeAndInputSize) =>
+    props.theme.input.borderColor};
+  border-radius: ${(props: PropsThemeAndInputSize) =>
+    props.theme.common[props.inputSize].borderRadius};
+  padding: ${(props: PropsThemeAndInputSize) =>
+    props.theme.common[props.inputSize].padding}
+  font-family: ${(props: PropsThemeAndInputSize) =>
+    props.theme.typography.fontFamily};
+  font-size: ${(props: PropsThemeAndInputSize) =>
+    props.theme.common[props.inputSize].fontSize}
+  color: ${(props: PropsThemeAndInputSize) => props.theme.reverseText};
+  &:disabled {
+    background: ${(props: PropsThemeAndInputSize) =>
+      props.theme.input.backgroundDisabled};
+    cursor: not-allowed;
+  }
+  .react-select-component > div {
+    border-color: ${(props: any) => props.invalid && 'red'};
```

Comment:
I used `any` here because I wasn't sure if we needed the prop types split as you have them. If the reason for splitting the types up is to handle the styling separately, it might be better to nest the styles for the error div in the styles for the wrapping div. That way we have access to all of the props throughout the tree.

Reply:
This makes sense, I should have done it this way!  Updating and consolidating...